packages/mobile/src/onboarding/contacts/ImportContactsScreen.tsx

```diff
@@ -1,227 +1,187 @@
-                {t('contacts.syncing.successHeader', { matches: matchesFound })}
-              </Text>
-              <Text style={styles.h2Status}>{t('contacts.syncing.successStatus')}</Text>
-            </>
-          ) : (
-            <Text style={styles.h1Status}>{t('contacts.syncing.successStatus')}</Text>
-          )}
-        </View>
-      )
+  // If import has already been done, move user forward
+  // Eventually we will want to somehow allow users to prompt matchmaking
+  React.useEffect(() => {
+    if (importStatus === ImportContactsStatus.Done) {
+      const onFinishTimeout = setTimeout(onFinish, 2000)
+      return () => {
+        clearTimeout(onFinishTimeout)
+      }
     }
+  }, [])
 
-    let h2Text: string
-    if (status === ImportContactsStatus.Processing) {
-      h2Text = 'contacts.syncing.loadingStatus1'
-    } else if (status === ImportContactsStatus.Matchmaking) {
-      h2Text = 'contacts.syncing.loadingStatus2'
-    } else {
-      h2Text = 'contacts.syncing.loadingStatus0'
+  React.useEffect(() => {
+    const prevImportContactProgress = prevImportContactProgressRef.current
+    const prevImportStatus = prevImportContactProgress.status
+    prevImportContactProgressRef.current = importContactsProgress
+
+    if (
+      prevImportStatus !== ImportContactsStatus.Done &&
+      importStatus === ImportContactsStatus.Done
+    ) {
+      const onFinishTimeout = setTimeout(onFinish, 1500)
+      return () => {
+        clearTimeout(onFinishTimeout)
+      }
     }
-    return (
-      <View style={styles.statusContainer}>
+  }, [importContactsProgress])
+
+  let statusComponents
+  if (importStatus === ImportContactsStatus.Done) {
+    statusComponents = (
+      <>
+        <Checkmark />
+        {matchedContactsCount > 0 ? (
+          <Text style={styles.statusText}>
+            {t('contacts.syncing.successWithMatches', { matches: matchedContactsCount })}
+          </Text>
+        ) : (
+          <Text style={styles.statusText}>{t('contacts.syncing.successWithoutMatches')}</Text>
+        )}
+      </>
+    )
+  } else {
+    statusComponents = (
+      <>
         <LoadingSpinner />
-        <Text style={styles.h1Status}>{t('contacts.syncing.loadingHeader')}</Text>
-        <Text style={styles.h2Status}>{t(h2Text, { total })}</Text>
-      </View>
+        <Text style={styles.statusText}>{t('contacts.syncing.loading')}</Text>
+      </>
     )
   }
 
-  render() {
-    const { isFindMeSwitchChecked } = this.state
-    const {
-      t,
-      importContactsProgress: { status },
-    } = this.props
-
-    const isSkipHidden = status === ImportContactsStatus.Done
-
-    return (
-      <SafeAreaView style={styles.container}>
-        <ScrollView contentContainerStyle={styles.scrollContainer}>
-          {status <= ImportContactsStatus.Stopped && (
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView contentContainerStyle={styles.scrollContainer}>
+        <View>
```

Comment:
nit: because we use fullscreen layout for onboarding screens, on iOS this makes the content collide with the navigation bar content when you overscroll,

<img src="https://user-images.githubusercontent.com/57791/85517055-2d542a80-b5ff-11ea-99a7-27df3c7008fc.png" width="50%" />

One way to prevent this is to start the scrollview under the navigation bar like done in:
https://github.com/celo-org/celo-monorepo/blob/aae2b147608ef1303df18b208c6778e01a22ccfd/packages/mobile/src/verify/VerificationEducationScreen.tsx#L68-L69

Reply:
I like this approach, thanks!